operator/src/config.rs

```diff
@@ -125,14 +156,122 @@ pub fn create_volumes(configmap_name: &str) -> Vec<Volume> {
 pub fn create_required_startup_env() -> Vec<EnvVar> {
     vec![
         EnvVar {
-            name: "SPARK_NO_DAEMONIZE".to_string(),
+            name: SPARK_NO_DAEMONIZE.to_string(),
             value: Some("true".to_string()),
             ..EnvVar::default()
         },
         EnvVar {
-            name: "SPARK_CONF_DIR".to_string(),
+            name: SPARK_CONF_DIR.to_string(),
             value: Some("{{configroot}}/conf".to_string()),
             ..EnvVar::default()
         },
     ]
 }
+
+/// Get all required configuration options
+/// 1) from spec
+/// 2) from node
+/// 3) from selector
+/// 4) from config properties
+/// # Arguments
+/// * `spec` - SparkCluster spec for common properties
+/// * `selector` - SparkClusterSelector containing desired config properties
+///
+pub fn get_config_properties(
+    spec: &SparkClusterSpec,
+    selector: &SparkNodeSelector,
+) -> HashMap<String, String> {
+    let mut config: HashMap<String, String> = HashMap::new();
+
+    let log_dir = &spec.log_dir.clone().unwrap_or_else(|| "/tmp".to_string());
+
+    // common
+    config.insert(SPARK_EVENT_LOG_ENABLED.to_string(), "true".to_string());
+    config.insert(SPARK_EVENT_LOG_DIR.to_string(), log_dir.to_string());
+
+    if let Some(secret) = &spec.secret {
+        config.insert(SPARK_AUTHENTICATE.to_string(), "true".to_string());
+        config.insert(SPARK_AUTHENTICATE_SECRET.to_string(), secret.to_string());
+    }
+
+    // history server
+    config.insert(
+        SPARK_HISTORY_FS_LOG_DIRECTORY.to_string(),
+        log_dir.to_string(),
+    );
+    if let Some(store_path) = &selector.store_path {
+        config.insert(SPARK_HISTORY_STORE_PATH.to_string(), store_path.to_string());
+    }
+    if let Some(port) = &selector.history_ui_port {
+        config.insert(SPARK_HISTORY_UI_PORT.to_string(), port.to_string());
+    }
+
+    // add config options -> may override previous settings which is what we want
```

Comment:
Can you elaborate which "config options" you mean here?
Clarify that these are the "free form" configs or whatever we want to call them